File: lc_soln/0167_two_sum_ii_input_array_is_sorted/solution.cs
public class Solution {
    public int[] TwoSum(int[] numbers, int target) {
        int l = 0;
        int r = numbers.Length - 1;
        while (true) {
            int diff = numbers[l] + numbers[r] - target;
            if (diff < 0) {
                l++;
            } else if (diff > 0) {
                r--;
            } else {
                return [l + 1, r + 1];
            }
        }
        return [];
    }
}

File: lc_soln/0167_two_sum_ii_input_array_is_sorted/solution.rs
struct Solution;

impl Solution {
    pub fn two_sum(numbers: Vec<i32>, target: i32) -> Vec<i32> {
        let mut l = 0;
        let mut r = numbers.len() - 1;
        loop {
            let diff = numbers[l as usize] + numbers[r as usize] - target;
            if diff < 0 {
                l += 1;
            } else if diff > 0 {
                r -= 1;
            } else {
                return vec![l as i32 + 1, r as i32 + 1];
            }
        }
        return vec![];
    }
}

fn main() {
    println!("{:?}", Solution::two_sum(vec![2, 7, 11, 15], 9));
}
